halo2_proofs/src/plonk/tachyon.rs

```diff
@@ -0,0 +1,441 @@
+use std::{
+    collections::{BTreeSet, HashMap},
+    io::Write,
+    ops::RangeTo,
+};
+
+use crate::{
+    bn254::{
+        AdviceSingle, Evals, InstanceSingle, ProvingKey as TachyonProvingKey, RationalEvals,
+        TachyonProver, TranscriptWriteState,
+    },
+    circuit::Value,
+    plonk::{
+        sealed, Advice, Any, Assigned, Assignment, Challenge, Circuit, Column, ConstraintSystem,
+        Error, Fixed, FloorPlanner, Instance, Selector,
+    },
+    poly::commitment::{Blind, CommitmentScheme},
+    transcript::{Challenge255, EncodedChallenge, Transcript, TranscriptWrite},
+    xor_shift_rng::XORShiftRng as TachyonXORShiftRng,
+};
+use ff::Field;
+use halo2curves::{
+    bn256::{Fr, G1Affine, G1},
+    group::{prime::PrimeCurveAffine, Curve},
+    CurveAffine,
+};
+
+/// This creates a proof for the provided `circuit` when given the public
```

Comment:
```suggestion
/// This creates a proof for the provided `circuits` when given the public
```

Reply:
Actually, since I just took this comment from Halo2, I'm worried his may produce another diffs !